DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -1451,24 +1453,26 @@ public async Task<DiscordMember> GetMemberAsync(ulong userId, bool updateCache =
     }
 
     /// <summary>
-    /// Retrieves a full list of members from Discord. This method will bypass cache.
+    /// Retrieves a full list of members from Discord. This method will bypass cache. Does one api request per 1000 member
     /// </summary>
+    /// <param name="cancellationToken">Cancels the enumeration before the next api request</param>
     /// <returns>A collection of all members in this guild.</returns>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async IAsyncEnumerable<DiscordMember> GetAllMembersAsync()
+    public async IAsyncEnumerable<DiscordMember> GetAllMembersAsync( [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
         int recievedLastCall = 1000;
         ulong last = 0ul;
         while (recievedLastCall > 0)
         {
-            IReadOnlyList<TransportMember> tms = await this.Discord.ApiClient.ListGuildMembersAsync(this.Id, 1000, last == 0 ? null : (ulong?)last);
+            if (cancellationToken.IsCancellationRequested) yield break;
+            IReadOnlyList<TransportMember> tms = await this.Discord.ApiClient.ListGuildMembersAsync(this.Id, 1000, last == 0 ? null : last);
             recievedLastCall = tms.Count;
 
             foreach (TransportMember transportMember in tms)
             {
                 DiscordUser user = new(transportMember.User) { Discord = this.Discord };
 
-                user = this.Discord.UpdateUserCache(user);
+                _ = this.Discord.UpdateUserCache(user);
```

Comment:
Does this method update the provided instance?

Reply:
```
internal DiscordUser UpdateUserCache(DiscordUser newUser)
        => this.UserCache.AddOrUpdate(newUser.Id, newUser, (_, _) => newUser);
``` 
This but the returned value isnt used anywhere